aries_cloudagent/core/tests/test_did_resolver.py

```diff
@@ -51,50 +53,108 @@
 test_resolver_1.supported_methods.return_value = [TEST_DID_METHOD1]
+test_resolver_1.native.return_value = False
+test_resolver_1.resolve.return_value = unittest.mock.MagicMock()
 test_resolver_2 = unittest.mock.MagicMock()
 test_resolver_2.supported_methods.return_value = [TEST_DID_METHOD2]
+test_resolver_2.native.return_value = False
+test_resolver_2.resolve.return_value = unittest.mock.MagicMock()
 test_resolver_3 = unittest.mock.MagicMock()
 test_resolver_3.supported_methods.return_value = [TEST_DID_METHOD3]
+test_resolver_3.native.return_value = False
+test_resolver_3.resolve.return_value = unittest.mock.MagicMock()
 test_resolver_4 = unittest.mock.MagicMock()
 test_resolver_4.supported_methods.return_value = [TEST_DID_METHOD4]
+test_resolver_4.native.return_value = False
+test_resolver_4.resolve.return_value = unittest.mock.MagicMock()
 TEST_RESOLVERS = [
     test_resolver_0,
     test_resolver_1,
     test_resolver_2,
     test_resolver_3,
     test_resolver_4,
 ]
-for resolver in resolvers:
+for resolver in TEST_RESOLVERS:
     did_resolver_registry.register(resolver)
 
 resolver = DIDResolver(did_resolver_registry)
 
+DOC = {
+    "@context": "https://w3id.org/did/v1",
+    "id": "did:example:1234abcd",
+    "verificationMethod": [
+        {
+            "id": "3",
+            "type": "RsaVerificationKey2018",
+            "controller": "did:example:1234abcd",
+            "publicKeyPem": "-----BEGIN PUBLIC X…",
+        },
+        {
+            "id": "4",
+            "type": "RsaVerificationKey2018",
+            "controller": "did:example:1234abcd",
+            "publicKeyPem": "-----BEGIN PUBLIC 9…",
+        },
+        {
+            "id": "6",
+            "type": "RsaVerificationKey2018",
+            "controller": "did:example:1234abcd",
+            "publicKeyPem": "-----BEGIN PUBLIC A…",
+        },
+    ],
+    "authentication": [
+        {
+            "type": "RsaSignatureAuthentication2018",
+            "publicKey": "did:example:1234abcd#4",
+        }
+    ],
+    "service": [
+        {
+            "id": "did:example:123456789abcdefghi;did-communication",
+            "type": "did-communication",
+            "priority": 0,
+            "recipientKeys": ["did:example:1234abcd#4"],
+            "routingKeys": ["did:example:1234abcd#3"],
+            "serviceEndpoint": "did:example:xd45fr567794lrzti67;did-communication",
+        }
+    ],
+}
 
 def test_create_resolver():
-    assert resolver.did_resolver_registery == TEST_RESOLVERS
+    assert len(resolver.did_resolver_registery.did_resolvers) == len(TEST_RESOLVERS)
 
 
 @pytest.mark.parametrize("did, method", zip(TEST_DIDS, TEST_DID_METHODS))
 def test_match_did_to_resolver(did, method):
-    base_resolver = resolver._match_did_to_resolver(did)
+    did = DID(did)
+    base_resolver = next(resolver._match_did_to_resolver(did))
     assert base_resolver.supports(method)
 
 
 def test_match_did_to_resolver_native_priority():
+    # TODO: implement this
```

Comment:
Using a pytest skip to mark tests that need more work is probably preferable to a todo comment